api/api/models_api.go

```diff
@@ -99,3 +104,126 @@ func (c *ModelsController) GetModel(r *http.Request, vars map[string]string, bod
 
 	return Ok(model)
 }
+
+// DeleteModel delete a model given a project and model ID
+func (c *ModelsController) DeleteModel(r *http.Request, vars map[string]string, body interface{}) *Response {
+	ctx := r.Context()
+
+	projectID, err := models.ParseID(vars["project_id"])
+	if err != nil {
+		return BadRequest("Unable to parse project_id")
+	}
+	modelID, err := models.ParseID(vars["model_id"])
+	if err != nil {
+		return BadRequest("Unable to parse model_id")
+	}
+
+	_, err = c.ProjectsService.GetByID(ctx, int32(projectID))
+	if err != nil {
+		return NotFound(err.Error())
+	}
+
+	model, err := c.ModelsService.FindByID(ctx, modelID)
+	if err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			return NotFound(fmt.Sprintf("Model is not found: %s", err.Error()))
+		}
+		return InternalServerError(err.Error())
+	}
+
+	// get all model version for this model
+	versions, _, err := c.VersionsService.ListVersions(ctx, modelID, c.MonitoringConfig, service.VersionQuery{})
+	if err != nil {
+		return InternalServerError(err.Error())
+	}
+
+	var allJobInModel [][]*models.PredictionJob
+	var allEndpointInModel [][]*models.VersionEndpoint
+
+	// iterate for every version, check the active prediction job / active endpoint
+	for _, version := range versions {
+		if model.Type == "pyfunc_v2" {
+			// check active prediction job
+			// if there are any active prediction job using the model version, deletion of the model version are prohibited
+			jobs, response := c.VersionsController.getInactivePredictionJobsForDeletion(ctx, model, version)
+			if response != nil {
+				return response
+			}
+
+			// save all prediction jobs for deletion process later on
+			allJobInModel = append(allJobInModel, jobs)
```

Comment:
Nit for renaming:
* `allJobInModel` -> `inactiveJobsInModel`
* `allEndpointInModel` -> `inactiveEndpointsInModel`
* `deleteInactivePredictionJobs` -> `deletePredictionJobs` (this function's logic is not checking whether the job is inactive, so it needn't be in the name)
* `deleteInactiveVersionEndpoints` -> `deleteVersionEndpoints` (same reasoning as above)

Reply:
Same comment on the Version API.